youtube_dl/extractor/brainpop.py

```diff
@@ -52,14 +52,14 @@ def _real_extract(self, url):
             'width': 480,
         })
         self._sort_formats(formats)
-        
-        settings = self._parse_json(self._html_search_regex(r'var settings = ([^;]*)', webpage, 'settings'), display_id)
+
+        settings = self._parse_json(self._html_search_regex(r'var settings = ([^;]*)', webpage, 'settings JSON', '{}'), display_id)
 
         return {
-            'id': content['category']['unit']['topic']['EntryID'],
+            'id': topic['EntryID'],
             'display_id': display_id,
-            'title': remove_end(settings['title'], ' - BrainPOP'),
-            'description': settings['description'],
+            'title': remove_end(settings.get('title', display_id), ' - BrainPOP'),
```

Comment:
`title` can also be extracted from `<title>(.*)</title>` also. One of them should be fatal, right?